File: src/main/java/com/range/rpms/dao/repository/UserRepository.java
package com.range.rpms.dao.repository;

import com.range.rpms.dao.model.User;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.Optional;

public interface UserRepository extends MongoRepository<User, String> {
    boolean existsByUsername(String username);

    boolean existsByEmail(String email);

    Optional<User> findByUsername(String username);
}

File: src/main/kotlin/com/example/rpms/users/domain/repository/UserRepository.kt
package com.example.rpms.users.domain.repository

import com.example.rpms.users.domain.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional
import java.util.UUID

interface UserRepository : JpaRepository<User, UUID> {
    fun findByUsername(username: String): User?
    fun findByEmail(email: String):User?
}